EXOSIMS/Prototypes/TargetList.py

```diff
@@ -1599,6 +1602,91 @@ def stellar_Teff(self):
             )
         )
 
+    def stellar_mass(self):
+        """Populates target list with 'true' and 'approximate' stellar masses
+
+        Approximate stellar masses are calculated from absolute magnitudes using the
+        model from [Henry1993]_. "True" masses are generated by a uniformly
+        distributed, 7%-mean error term to the apprxoimate masses.
+
+        All values are in units of solar mass.
+
+        Function called by reset sim.
+
+        """
+
+        if self.massLuminosityRelationship == "Henry1993":
+            # good generalist, but out of date
+            # 'approximate' stellar mass
+            self.MsEst = (
+                10.0 ** (0.002456 * self.MV**2 - 0.09711 * self.MV + 0.4365)
+            ) * u.solMass
+            # normally distributed 'error' of 7%
+            err = (np.random.random(len(self.MV)) * 2.0 - 1.0) * 0.07
+            self.MsTrue = (1.0 + err) * self.MsEst
+
+        elif self.massLuminosityRelationship == "Fernandes2021":
+            # only good for FGK
+            # 'approximate' stellar mass without error
+            self.MsEst = (
+                10
+                ** (
+                    (0.219 * np.log10(self.L))
+                    + (0.063 * ((np.log10(self.L)) ** 2))
+                    - (0.119 * ((np.log10(self.L)) ** 3))
+                )
+            ) * u.solMass
+            # error distribution in literature as 3% in approxoimate masses
+            err = (np.random.random(len(self.L)) * 2.0 - 1.0) * 0.03
+            self.MsTrue = (1.0 + err) * self.MsEst
+
+        elif self.massLuminosityRelationship == "Henry1993+1999":
+            # more specific than Henry1993
+            # initialize MsEst attribute
+            self.MsEst = np.array([])

```

Comment:
you should try to avoid 'growing' arrays like this whenever possible.  You know the exact final size this array should be - allocate the whole thing with `np.zeros` and then index it directly to populate.